### PR TITLE
update readme

### PR DESCRIPTION
**Pull request by bucko13**
Clearer language around the different setup options, docker, configuration, etc. Added a TOC for easier navigation and a bird's eye view of the content.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -35,87 +48,47 @@ npm start -- --client-id=main
...
-necessary constants.
-1. Run `docker-compose up -d` to start everything.
-1. Navigate to [localhost:5000](http://localhost:5000) to see your webapp.
-1. Or navigate to [https://localhost](https://localhost) to use TLS - you will have to
-choose to trust the certificate
+## Managing Plugins
 
-Requests to the `/bcoin` will get forwarded to your bcoin node.
-Requests to the `/bwallet` endpoint will get forwarded to the wallet server
+Plugin management is handled in a `js` file located by default in the `.bpanel` directory in your
```
Comment (pinheadmz) [inline README.md]: mention the file's name here too?
> located by default at `.bpanel/config.js`

File: README.md [hunk 2]
```diff
@@ -35,87 +48,47 @@ npm start -- --client-id=main
...
+bPanel comes pre-installed with a default theme called [`Genesis Theme`](https://github.com/bpanel-org/genesis-theme)
+that bundles together a set of useful starter plugins and a custom skin called [bMenace](https://github.com/bpanel-org/bmenace-theme).
 If you want, you can disable the Genesis Theme by removing it from the list in `config.js`,
 but if you want to keep using _some_ of the plugins from the theme, feel free to add
 them individually to your config!
 
+### Installing Plugins
+
 To install plugins, simply add the name as a string to the `plugins` array in `~/.bpanel/config.js`.
 Make sure to match the name to the package name on npm
```
Comment (pinheadmz) [inline README.md]: example might be helpful here to show punctuation, formatting:

>   plugins: ['@bpanel/bmoon-theme', '@bpanel/publish-data']

File: README.md [hunk 3]
```diff
@@ -156,54 +130,80 @@ BPANEL_TLS_CERT=<path to cert>
...
 out of the box if you simply run `docker-compose up -d`.
 
 Configurations are shared between the two docker containers using a shared
 docker volume called `configs`. Settings for the bcoin nodes in docker
 are set using environment variables, either in docker-compose.yml or
-an env file (by default in `/etc/bcoin.env` but you can point to whichever and
+an env file (by default in `./etc/regtest.bcoin.env` but you can point to whichever and
 as many env files as you want using the `env_file` configuration in the bcoin
 service). The bcoin node is started with the `bcoin-init.js` script. During this
 process, api keys are generated and all required configurations are saved in a config
```
Comment (pinheadmz) [inline README.md]: nit: API

File: README.md (hunk 2 above)
Comment (bucko13) [reply README.md]: Added an example. FWIW, the formatting/punctuation around the `@` for example is still strictly the plugin name. i.e. whatever you have to type for `npm install` is what should be written in the array. Our plugins just happen to be scoped on npm under `@bpanel`

File: README.md (hunk 1 above)
Comment (bucko13) [reply README.md]: updated

File: README.md (hunk 3 above)
Comment (bucko13) [reply README.md]: 👍

File: README.md [hunk 4]
```diff
@@ -35,87 +48,58 @@ npm start -- --client-id=main
...
+### With Docker
+
 From inside the project directory in your terminal, run:
 
 ```bash
 npm install && docker-compose up -d
 ```
 
-This will start up three docker services as background daemon processes.
+This will start up three docker services as [detached processes running in the background](https://docs.docker.com/compose/reference/up/).
```
Comment (tynes) [inline README.md]: Services are a thing in docker - https://docs.docker.com/get-started/part3/#about-services

We aren't specifically spinning up services, but containers. Maybe:

`this will start up docker containers as...`

File: README.md [hunk 5]
```diff
@@ -35,87 +48,58 @@ npm start -- --client-id=main
...
 
-1. Run `npm install` to install dependencies and generate
-necessary constants.
-1. Run `docker-compose up -d` to start everything.
-1. Navigate to [localhost:5000](http://localhost:5000) to see your webapp.
-1. Or navigate to [https://localhost](https://localhost) to use TLS - you will have to
-choose to trust the certificate
+Plugin management is handled in a `config.js` file located by default in the `.bpanel` directory in your
+system's home directory (this location can be manually changed with the `BPANEL_PREFIX` env variable or
+`--prefix` option at run time.
```
Comment (tynes) [inline README.md]: great comment

File: README.md [hunk 6]
```diff
@@ -156,54 +141,80 @@ BPANEL_TLS_CERT=<path to cert>
...
-If the configs volume is mounted to your host machine, you can connect a local
+#### Local bPanel Dev (non-docker) with bcoin docker service
+
+If the configs volume is mounted and mapped to your host machine, you can connect a local
 version of bPanel to it by pointing the `client-id` config at `_docker` and it will
 use the appropriate configs.
 
+For local development, you run just the bcoin docker container (`docker-compose up -d bcoin`)
+and then `npm run start:dev -- --client-id=_docker` (or `npm run start:poll -- --client-id=_docker`
+for Mac since webpack's watch can behave strangely on macOS) to run bPanel and its server
```
Comment (tynes) [inline README.md]: `since webpack's watch can behave strangely on macOS` maybe reword this sentence

```
(or `npm run start:poll -- --client-id=_docker`
on Mac to enbable webpack's watch with filesystem polling)
```
